internal/docs/readme.go

```diff
@@ -258,3 +264,78 @@ func readmePath(fileName, packageRoot string) string {
 func docsPath(packageRoot string) string {
 	return filepath.Join(packageRoot, "docs")
 }
+
+func ValidateDocsStructure(packageRoot string) error {
+	logger.Debugf("Validating docs structure (package: %s)", packageRoot)
+
+	bm, ok, err := buildmanifest.ReadBuildManifest(packageRoot)
+	if err != nil {
+		return fmt.Errorf("reading build manifest failed: %w", err)
+	}
+	if !ok {
+		return fmt.Errorf("build manifest not found in package root: %s", packageRoot)
+	}
+
+	if !bm.Dependencies.DocsStructureEnforced {
+		logger.Debug("Docs structure validation is not enforced, skipping")
+		return nil
+	}
+
+	logger.Debug("Docs structure validation is enforced, proceeding with validation")
+
+	err = validateReadmeStructure(packageRoot)
+	if err != nil {
+		return fmt.Errorf("structure validation: %w", err)
+	}
+
+	return nil
+}
+
+func validateReadmeStructure(packageRoot string) (error) {
+	requiredHeaders := []string{"Overview", "Troubleshooting", "Scaling"}
+	fileName := "README.md"
```

Comment:
There are packages with a main README.md, and multiple .md files, one for each policy template. An example of this is the AWS package. Do we want to apply the same validation in these cases?